service/src/main/java/bio/terra/workspace/app/controller/ControlledAzureResourceApiController.java

```diff
@@ -741,4 +746,102 @@ public ResponseEntity<ApiAzureManagedIdentityResource> getAzureManagedIdentity(
             .castByEnum(WsmResourceType.CONTROLLED_AZURE_MANAGED_IDENTITY);
     return new ResponseEntity<>(resource.toApiResource(), HttpStatus.OK);
   }
+
+  @Override
+  public ResponseEntity<ApiCreatedControlledAzureKubernetesNamespaceResult>
+      createAzureKubernetesNamespace(
+          UUID workspaceId, ApiCreateControlledAzureKubernetesNamespaceRequestBody body) {
+    features.azureEnabledCheck();
+
+    final AuthenticatedUserRequest userRequest = getAuthenticatedInfo();
+    final ControlledResourceFields commonFields =
+        toCommonFields(
+            workspaceId,
+            body.getCommon(),
+            landingZoneApiDispatch.getLandingZoneRegion(
+                userRequest, workspaceService.getWorkspace(workspaceId)),
+            userRequest,
+            WsmResourceType.CONTROLLED_AZURE_DATABASE);
+    workspaceService.validateMcWorkspaceAndAction(
+        userRequest, workspaceId, ControllerValidationUtils.samCreateAction(commonFields));
+    workspaceService.validateWorkspaceAndContextState(workspaceId, CloudPlatform.AZURE);
+
+    // the namespace must be unique across all workspaces in the LZ
```

Comment:
What verifies/enforces that it is unique?

Reply:
this is the reason the next line appends workspaceId, I will make the comment more clear